### PR TITLE
Move scaling to KTrajectory

### PR DESCRIPTION
**Pull request by fzimmermann89**
There is a bit of duplicaiton for KTrajectoryRawShape and KTrajectory, but inherting one from the other makes KData more complicated. I might remove this duplication in #560 

I also added the ability to load in Trajectories with the wrong, i.e. not zyx, order.

### CODE REVIEW

**Comment by fzimmermann89** [comment]
Would be nice to get this in before the release @schuenke

File: src/mrpro/data/KTrajectory.py [hunk 1]
```diff
@@ -85,13 +89,33 @@ def from_tensor(
...
             tolerance to detect if trajectory points are on integer grid positions
+        encoding_matrix
+            if an encoding matrix is supplied, the trajectory is rescaled to fit
+            within the matrix. Otherwise, it is left as-is.
         """
-        kz, ky, kx = torch.unbind(tensor, dim=stack_dim)
+        ks = tensor.unbind(dim=stack_dim)
+        kz, ky, kx = (ks[axes_order.index(axis)] for axis in 'zyx')
+
+        def normalize(k: torch.Tensor, encoding_size: int) -> torch.Tensor:
```
Comment (schuenke) [inline src/mrpro/data/KTrajectory.py]: rename normalize -> rescale / scale etc

File: src/mrpro/data/KTrajectory.py [hunk 2]
```diff
@@ -85,13 +89,33 @@ def from_tensor(
...
         stack_dim
             The dimension in the tensor the directions have been stacked along.
+        axes_order
+            Order of the axes in the tensor. Our convention usually is zyx order.
         repeat_detection_tolerance
             detects if broadcasting can be used, i.e. if dimensions are repeated.
             Set to None to disable.
         grid_detection_tolerance
             tolerance to detect if trajectory points are on integer grid positions
+        encoding_matrix
```
Comment (schuenke) [inline src/mrpro/data/KTrajectory.py]:
```suggestion
        scaling_matrix
```

File: src/mrpro/data/KTrajectory.py [hunk 3]
```diff
@@ -85,13 +89,33 @@ def from_tensor(
             This should be a 5-dim tensor, with (kz,ky,kx) stacked in this order along stack_dim
         stack_dim
             The dimension in the tensor the directions have been stacked along.
+        axes_order
+            Order of the axes in the tensor. Our convention usually is zyx order.
         repeat_detection_tolerance
             detects if broadcasting can be used, i.e. if dimensions are repeated.
             Set to None to disable.
```
Comment (schuenke) [inline src/mrpro/data/KTrajectory.py]:
```suggestion
        repeat_detection_tolerance
            Tolerance for detecting repeated dimensions (broadcasting).
            If trajectory points differ by less than this value, they are considered
            identical. Set to None to disable this feature.
```

File: src/mrpro/data/KTrajectory.py [hunk 4]
```diff
@@ -85,13 +89,33 @@ def from_tensor(
...
         grid_detection_tolerance
             tolerance to detect if trajectory points are on integer grid positions
```
Comment (schuenke) [inline src/mrpro/data/KTrajectory.py]:
```suggestion
            Tolerance for detecting whether trajectory points align with integer 
            grid positions. This tolerance is applied after rescaling if 
            `scaling_matrix` is provided.
```

File: src/mrpro/data/KTrajectory.py [hunk 5]
```diff
@@ -85,13 +89,33 @@ def from_tensor(
...
+        encoding_matrix
+            if an encoding matrix is supplied, the trajectory is rescaled to fit
+            within the matrix. Otherwise, it is left as-is.
```
Comment (schuenke) [inline src/mrpro/data/KTrajectory.py]:
```suggestion
            If a scaling matrix is provided, the trajectory is rescaled to fit within 
            the dimensions of the matrix. If not provided, the trajectory remains unchanged.

```

Comment (schuenke) [reply src/mrpro/data/KTrajectory.py]: and maybe add an example like
`i.e. SpatialDimension(z=1, y=64, x=64) for a single slice acquisition with 64x64 pixels`

**Review by schuenke — APPROVED** [review]
See my suggestions as discussed before. Use same/similar docstrings in other class as well. 
But code works nice. Tested with the MAGMA review notebook!

File: src/mrpro/data/KTrajectory.py [hunk 6]
```diff
@@ -85,13 +89,33 @@ def from_tensor(
...
+                # a single encoding point should be at zero
+                # avoid division by zero
+                return torch.zeros_like(k)
+            return k * (encoding_size / max_abs_range)
+
+        if encoding_matrix is not None:
+            kz = normalize(kz, encoding_matrix.z)
+            ky = normalize(ky, encoding_matrix.y)
+            kx = normalize(kx, encoding_matrix.x)
+
```
Comment (fzimmermann89) [inline src/mrpro/data/KTrajectory.py]:
```suggestion
        def rescale(k: torch.Tensor, size: float) -> torch.Tensor:
            max_abs_range = 2 * k.abs().max()
            if size < 2 or max_abs_range < 1e-6:
                # a single encoding point should be at zero
                # avoid division by zero
                return torch.zeros_like(k)
            return k * (size / max_abs_range)

        if scaling_matrix is not None:
            kz = rescale(kz, scaling_matrix.z)
            ky = rescale(ky, scaling_matrix.y)
            kx = rescale(kx, scaling_matrix.x)

```
